### PR TITLE
Rework how rollups are created

### PR DESCRIPTION
**Pull request by StefanIliev545**
### Why this change is needed

This change makes rollups more reliable and removes the concept of needing them to be linked or having anything to do with each other. The host controls from which batch the rollup is continued and this information is taken from the l1. 

### What changes were made as part of this PR

Reworked rollup producer and consumer, the management contract and in general how things are stored/read. 

### PR checks pre-merging

Please indicate below by ticking the checkbox that you have read and performed the required
[PR checks](https://github.com/obscuronet/obscuro-internal/blob/main/dev-ops-docs/dev-pr-checks.md)

- [ ] PR checks reviewed and performed

### CODE REVIEW

File: contracts/src/management/ManagementContract.sol [hunk 1]
```diff
@@ -15,125 +15,39 @@ contract ManagementContract {
...
-        tree.rollupChildren[_parentID].push(rollupID);
-
-        // store the hashpointer
-        tree.rollupsHashes[_r.Hash] = rollupID;
-
-        // mark this as the head
-        if (parent.ElementID == tree._HEAD) {
-            tree._HEAD = rollupID;
-        }
+    function GetHeadRollup() internal view returns ( Structs.MetaRollup memory ) {
```
Comment (tudor-malene) [inline contracts/src/management/ManagementContract.sol:40]: why is there a "head rollup" if there is no chain?

File: go/enclave/components/rollup_consumer.go [hunk 2]
```diff
@@ -71,19 +69,14 @@ func (rc *rollupConsumerImpl) ProcessL1Block(b *common.BlockAndReceipts) (*commo
 		return nil, nil //nolint:nilnil
 	}
 
-	rollup, err := rc.getCanonicalRollup(rollups, b)
-	if err != nil {
-		return nil, err
-	}
-
-	err = rc.processRollup(b.Block, rollup)
+	rollup, err := rc.getSignedRollup(rollups, b)
```
Comment (tudor-malene) [inline go/enclave/components/rollup_consumer.go:72]: I think you need to get all signed rollups and process them one by one as if they came independently.

File: go/host/l1/publisher.go [hunk 3]
```diff
@@ -170,6 +173,23 @@ func (p *Publisher) ExtractSecretResponses(block *types.Block) []*ethadapter.L1R
 	return secretRespTxs
 }
 
+func (p *Publisher) FetchLatestSeqNo() (*big.Int, error) {
```
Comment (tudor-malene) [inline go/host/l1/publisher.go:176]: good idea to read the last sequence number from the actual contract, as the ultimate source of truth

**Review by tudor-malene — COMMENTED** [review]
looks good.
One minor comment.

File: contracts/src/management/ManagementContract.sol (hunk 1 above)
Comment (StefanIliev545) [reply contracts/src/management/ManagementContract.sol:40]: I'm not sure if this is needed at this point. Technically there is a "sequence" of rollups with ever increasing batch sequence numbers so there is a head rollup. I will remove it given that its internal and not really used anywhere.

**Review by tudor-malene — APPROVED** [review]
lgtm (once tests pass)

**Pull request merged**